File: app/src/main/java/com/example/worktracker/ActivityTrackActivity.java
package com.example.worktracker;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.ArrayList;

public class ActivityTrackActivity extends AppCompatActivity {
    private TextView status;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_activitytrack);

        // Initialize Firebase
        FirebaseDatabase database = FirebaseDatabase.getInstance("https://scl-filipinas-work-tracker-default-rtdb.asia-southeast1.firebasedatabase.app/a"); // Replace with your database URL
        DatabaseReference activitiesRef = database.getReference("Activities");
        DatabaseReference activityTypeRef = database.getReference("ActivityTypes");
        DatabaseReference cropRef = database.getReference("Crops");

        // UI Elements
        Button goBackBtn = findViewById(R.id.goBackBtn);
        Button saveBtn = findViewById(R.id.saveBtn);
        EditText dateField = findViewById(R.id.dateField);
        Spinner activityTypeField = findViewById(R.id.activityTypeField);
        Spinner targetCropsField = findViewById(R.id.targetCropsField);
        EditText attendanceField = findViewById(R.id.attendanceField);
        EditText activityCostField = findViewById(R.id.activityCostField);
        status = findViewById(R.id.status);

        // Fetch and display data
        fetchAndDisplayData(activityTypeRef, activityTypeField);
        fetchAndDisplayData(cropRef, targetCropsField);

        // Back button
        goBackBtn.setOnClickListener(v -> {
            Intent intent = new Intent(getApplicationContext(), MainActivity.class);
            startActivity(intent);
            finish();
        });

        // Save button
        saveBtn.setOnClickListener(v -> {
            String date = dateField.getText().toString();
            String activityType = activityTypeField.getSelectedItem().toString();
            String targetCrop = targetCropsField.getSelectedItem().toString();
            String attendance = attendanceField.getText().toString();
            String activityCost = activityCostField.getText().toString();

            if (date.isEmpty() || activityType.isEmpty() || targetCrop.isEmpty() || attendance.isEmpty() || activityCost.isEmpty()) {
                status.setText(getString(R.string.empty));
            } else {
                activitiesRef.push().setValue(date + "," + activityType + "," + targetCrop + "," + attendance + "," + activityCost);
                status.setText(getString(R.string.saved));
            }
        });
    }

    private void fetchAndDisplayData(DatabaseReference ref, Spinner spinner) {
        ref.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                ArrayList<String> values = new ArrayList<>();
                for (DataSnapshot snapshot : dataSnapshot.getChildren()) {
                    values.add(snapshot.getValue(String.class)); // Assumes the data is of type String
                }
                ArrayAdapter<String> adapter = new ArrayAdapter<>(ActivityTrackActivity.this,
                        android.R.layout.simple_list_item_1, values);
                spinner.setAdapter(adapter);
            }

            @Override
            public void onCancelled(@NonNull DatabaseError databaseError) {
                // Failed to read value
                status.setText("Failed to load data");
            }
        });
    }
}

File: app/src/main/java/com/example/worktracker/ActivityTrackActivity.kt
package com.example.worktracker

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class ActivityTrackActivity : AppCompatActivity() {
    private lateinit var goBackBtn: Button
    private lateinit var saveBtn: Button
    private lateinit var dateField: EditText
    private lateinit var activityTypeField: EditText
    private lateinit var targetCropsField: EditText
    private lateinit var attendanceField: EditText
    private lateinit var activityCostField: EditText
    private lateinit var status: TextView

    public override fun onStart() {
        super.onStart()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        // just for appearance
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        setContentView(R.layout.activity_activitytrack)

        goBackBtn = findViewById(R.id.goBackBtn)
        saveBtn = findViewById(R.id.saveBtn)
        dateField = findViewById(R.id.dateField)
        activityTypeField = findViewById(R.id.activityTypeField)
        targetCropsField = findViewById(R.id.targetCropsField)
        attendanceField = findViewById(R.id.attendanceField)
        activityCostField = findViewById(R.id.activityCostField)
        status = findViewById(R.id.status)

        // back button
        goBackBtn.setOnClickListener {
            val intent = Intent(applicationContext, MainActivity::class.java)
            startActivity(intent)
            finish()
        }

        // save button
        saveBtn.setOnClickListener {
            //TODO: Append to .csv file

            // concatenate input values
            val outputStr = "${dateField.text}," +
                    "${activityTypeField.text}," +
                    "${targetCropsField.text}," +
                    "${attendanceField.text}," +
                    "${activityCostField.text}"
            status.text = getString(R.string.saved)
        }

        // just for appearance
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }
}